hw/ip/otbn/doc/_index.md

```diff
@@ -439,6 +439,25 @@ outer_body:
 
 ## Design Details
 
+### Memories
+
+The OTBN processor core has access to two dedicated memories:
+an instruction memory (IMEM), and a data memory (DMEM).
```

Comment:
Mega-nit: If we're doing one sentence per line, I think these should be combined. Similarly for the IMEM paragraph a couple of lines below.